server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -236,6 +235,16 @@ public int add(InputRow row, Supplier<Committer> committerSupplier) throws Index
       persist(committerSupplier.get());
     }
 
+    if (++counter % CHECK_INTERNAL_COUNT == 0) {
+      int size = 0;
+      for (Sink aSink : sinks.values()) {
+        size += aSink.size();
+      }
+      if (size > config.getMaxRowsInMemory()) {
+        persistOldest(committerSupplier.get());
```

Comment:
Yes, I've thought on that. Why I did like this is that it's realtime indexer not wanting persist actively ingesting index.